x-pack/plugins/security_solution/public/common/components/autocomplete/field_value_match.tsx

```diff
@@ -98,27 +105,72 @@ export const AutocompleteFieldMatchComponent: React.FC<AutocompleteFieldMatchPro
     isLoadingSuggestions,
   ]);
 
-  return (
-    <EuiComboBox
-      placeholder={inputPlaceholder}
-      isDisabled={isDisabled}
-      isLoading={isLoadingState}
-      isClearable={isClearable}
-      options={comboOptions}
-      selectedOptions={selectedComboOptions}
-      onChange={handleValuesChange}
-      singleSelection={{ asPlainText: true }}
-      onSearchChange={onSearchChange}
-      onCreateOption={onChange}
-      isInvalid={!isValid}
-      onFocus={setIsTouchedValue}
-      sortMatchesBy="startsWith"
-      data-test-subj="valuesAutocompleteComboBox matchComboxBox"
-      style={fieldInputWidth ? { width: `${fieldInputWidth}px` } : {}}
-      fullWidth
-      async
-    />
-  );
+  const getDefaultInput = (): JSX.Element => {
```

Comment:
I'm a little rusty on the component render optimization front, but I believe you can memoize this function, and wrap the above `handler` functions in a `useCallback` to maintain references and minimize some unnecessary renders. 

I'm seeing ~5 renders per entry when opening the exceptions modal/creating a new entry, but it may just be the parents props causing this. The UI still  _feels snappy_, and that's always a good litmus test, but just wanted to give a heads up so no changes necessary here. There will be plenty of fun stuff to uncover when you go on your perf adventures 🙂

Reply:
Updated!